drivers/focuser/ieaffocus.cpp

```diff
@@ -0,0 +1,709 @@
+        return true;
+    }
+
+    return INDI::Focuser::ISNewNumber(dev, name, values, names, n);
+
+
+
+
+}
+
+void iEAFFocus::GetFocusParams ()
+{
+    if (updateTemperature())
+        TemperatureNP.apply();
+
+    if (updateMaxPos())
+    {
+        MaxPositionNP.apply();
+        IDSetNumber(&FocusAbsPosNP, nullptr);
+    }
+
+
+    readReverseDirection();
+//    updatePosition();
+
+
+    if (updatePosition())
+        IDSetNumber(&FocusAbsPosNP, nullptr);
+
+}
+
+
+IPState iEAFFocus::MoveAbsFocuser(uint32_t targetTicks)
+{
+    uint32_t targetPos = targetTicks;
+
+    bool rc = false;
+
+    rc = MoveMyFocuser(targetPos);
+
+    if (rc == false)
+        return IPS_ALERT;
+
+    FocusAbsPosNP.s = IPS_BUSY;
+
+    return IPS_BUSY;
+}
+
+IPState iEAFFocus::MoveRelFocuser(FocusDirection dir, uint32_t ticks)
+{
+    uint32_t newPosition = 0;
+    bool rc = false;
+
+    if (dir == FOCUS_INWARD)
+        newPosition = uint32_t(FocusAbsPosN[0].value) - ticks;
+    else
+        newPosition = uint32_t(FocusAbsPosN[0].value) + ticks;
+
+    rc = MoveMyFocuser(newPosition);
+
+    if (rc == false)
+        return IPS_ALERT;
+
+    FocusRelPosN[0].value = ticks;
+    FocusRelPosNP.s = IPS_BUSY;
+
+    return IPS_BUSY;
+}
+
+void iEAFFocus::TimerHit()
+{
+    if (isConnected() == false)
+    {
+        SetTimer(POLLMS_OVERRIDE);
+        return;
+    }
+
+    bool rc = updatePosition();
+    if (rc)
+    {
+        if (fabs(lastPos - FocusAbsPosN[0].value) > 5)
+        {
+            IDSetNumber(&FocusAbsPosNP, NULL);
+            lastPos = FocusAbsPosN[0].value;
+        }
+    }
+
+
+    if (FocusAbsPosNP.s == IPS_BUSY || FocusRelPosNP.s == IPS_BUSY)
+    {
+        if (isMoving() == false)
+        {
+            FocusAbsPosNP.s = IPS_OK;
+            FocusRelPosNP.s = IPS_OK;
+            IDSetNumber(&FocusAbsPosNP, NULL);
+            IDSetNumber(&FocusRelPosNP, NULL);
+            lastPos = FocusAbsPosN[0].value;
+            DEBUG(INDI::Logger::DBG_SESSION, "Focuser reached requested position.");
+        }
+    }
+    SetTimer(POLLMS_OVERRIDE);
+
+}
+
+bool iEAFFocus::AbortFocuser()
+{
+    int nbytes_written;
+    if (tty_write(PortFD, ":FQ#", 4, &nbytes_written) == TTY_OK)
+    {
+        FocusAbsPosNP.s = IPS_IDLE;
+        FocusRelPosNP.s = IPS_IDLE;
+        IDSetNumber(&FocusAbsPosNP, NULL);
+        IDSetNumber(&FocusRelPosNP, NULL);
+        return true;
+    }
+    else
+        return false;
+}
+
+
+bool iEAFFocus::saveConfigItems(FILE * fp)
+{
+    Focuser::saveConfigItems(fp);
+/*
+    IUSaveConfigNumber(fp, &TemperatureSettingNP);
+    IUSaveConfigSwitch(fp, &TemperatureCompensateSP);
+    IUSaveConfigSwitch(fp, &BacklashInSP);
+    IUSaveConfigNumber(fp, &BacklashInStepsNP);
+    IUSaveConfigSwitch(fp, &BacklashOutSP);
+    IUSaveConfigNumber(fp, &BacklashOutStepsNP);
+    IUSaveConfigSwitch(fp, &StepModeSP);
+    IUSaveConfigSwitch(fp, &DisplaySP);
+*/
+    //IUSaveConfigSwitch(fp, &DisplaySP);
+    return true;
+}
```

Comment:
If nothing is implemented here, then no need to define the function. It can be safely removed.